server/src/main/java/org/elasticsearch/common/lucene/store/InputStreamIndexInput.java

```diff
@@ -52,8 +52,8 @@ public int read(byte[] b, int off, int len) throws IOException {
         if (counter + len > limit) {
             len = (int) (limit - counter);
         }
-        if (len <= 0) {
-            return -1;
+        if (len == 0) {
```

Comment:
What if `len < -1` here?

Reply:
`IndexOutOfBoundsException` would be thrown. There is the `off < 0 || len < 0 || len > b.length - off` condition a couple of lines before. I reverted the condition to `len <= 0` for the consistency sake, though.

Reply:
If `len < 0` is impossible then IMO it would be helpful to future readers to assert that `len >= 0` here.